### ISSUE TITLE
Vagrant Cloud post-processor upload errors with unexpected EOF

### ISSUE THREAD

**Issue by Peleke** [hashicorp/packer#8331]
#### Overview of the Issue

tl;dr: Vagrant Cloud post-processor fails with an unexpected EOF when uploading a successful build targeting Virtual Box. The build is running on a Jenkins agent. I'm running the build locally at the moment, and will report back with results when it finishes.

I'm using a near-minimal Packer template with the Vagrant builder to run an Ansible build on the `bento/ubuntu-18.04`  base box. Template is attached below. I've stripped it down to the bare essentials and reduced the Ansible provisioning to a series of package installs with `apt` to attempt to pinpoint the error, so there aren't many distractions.

#### Reproduction Steps

Trigger the build on a Jenkins agent via Blue Ocean pipelines. 

See infrastructure details below for details.

### Packer version

Packer version `1.4.4`.

### Simplified Packer Buildfile

<details>
<summary><b>Packer template</b> (`UbuntuVM.json`)</summary>
<pre>
{
    "variables": {
        "cloud_token": "{{ env `VAGRANT_CLOUD_TOKEN` }}",
        "version": "1.0.{{timestamp}}"
    },
    "builders": [{
        "type": "vagrant",
        "communicator": "ssh",
        "provider": "virtualbox",
        "source_path": "bento/ubuntu-18.04",
        "add_force": true
    }],
    "provisioners": [{
        "type": "ansible",
        "playbook_file": "../provisioners/main.yml"
    }],
    "post-processors": [
        {
            "type": "vagrant-cloud",
            "box_tag": "cybersecurity/UbuntuVM",
            "access_token": "{{user `VAGRANT_CLOUD_TOKEN`}}",
            "version": "{{user `version`}}"
        }
    ]
}
</pre>
</details>

<details>
<summary><b>Jenkinsfile</b></summary>
<pre>
pipeline {
  agent any
  stages {
    stage('Build') {
      steps {
        sh(label: 'Check directory', script: 'pwd && whoami')
        sh(label: 'Verify Packer is on $PATH', script: '/usr/local/bin/packer --version')
      }
    }

    stage('Packer') {
      steps {
        echo 'Packing...'
        sh(label: 'Packer Build Stage!', script: '/usr/local/bin/packer build -debug UbuntuVM/packer/UbuntuVM.json')
      }
    }

    stage('Deploy') {
      steps {
        echo 'Deploying...'
      }
    }

  }
  environment {
    VAGRANT_CLOUD_TOKEN = 'Omitted Cause...Secrets'
  }
}
</pre>
</details>

### Operating system and Environment details

**Infrastructure**
- Jenkins v2.203 running on Ubuntu 18.04, deployed to an Azure `Standard_D4S_v3` instance
- The Ubuntu instance is unmodified _except_ for the following installations:
  - Vagrant v2.0.2
  - Ansible v2.9.0
  - Packer v1.4.4

**Build Details**
- The build is described in a declarative `Jenkinsfile` — see above.
- Fwiw, webhooks are turned off at the moment, and the build is triggered manually.

Builds, for now, are run directly on the Jenkins master as opposed to on distributed agents or within Docker containers.

### Log Fragments and crash.log files

See the [gist](https://gist.github.com/Peleke/daea9cb95c4003f6440a35feb095b9c5) for full log output.

### Additional Details / Some Questions

The error message I get is:

```
Error uploading box! Will retry in 10 seconds. Error: Put https://archivist.vagrantup.com/v1/object/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJib3hlcy82MTBlYTY1OS1kNWRlLTQ2MjctODczNy0yY2I3MjRlNjIxMzkiLCJtb2RlIjoidyIsImV4cGlyZSI6MTU3Mjk3Mzc3NiwiY2FsbGJhY2siOiJodHRwczovL3ZhZ3JhbnRjbG91ZC5jb20vYXBpL2ludGVybmFsL2FyY2hpdmlzdC9jYWxsYmFjayJ9.s5DFfIzBBMFo-4MzMy4SXYcOZ0v79pZqea5a0RSbfqQ: EOF[0m
```

...Which is thrown from [packer/post-processor/vagrant-cloud/step_upload.go](https://github.com/hashicorp/packer/blob/dcd21d37da053ccb2fc194c77e710ab6eb5967bf/post-processor/vagrant-cloud/step_upload.go#L17). Unfortunately, I get no additional details regarding the nature of the error:

```
2019/11/05 16:57:15 packer: 2019/11/05 16:57:15 Post-Processor Vagrant Cloud API Upload: output-vagrant/package.box https://archivist.vagrantup.com/v1/object/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJib3hlcy82MTBlYTY1OS1kNWRlLTQ2MjctODczNy0yY2I3MjRlNjIxMzkiLCJtb2RlIjoidyIsImV4cGlyZSI6MTU3Mjk3Mzc3NiwiY2FsbGJhY2siOiJodHRwczovL3ZhZ3JhbnRjbG91ZC5jb20vYXBpL2ludGVybmFsL2FyY2hpdmlzdC9jYWxsYmFjayJ9.s5DFfIzBBMFo-4MzMy4SXYcOZ0v79pZqea5a0RSbfqQ

2019/11/05 16:58:25 packer: 2019/11/05 16:58:25 Post-Processor Vagrant Cloud Upload Response:

2019/11/05 16:58:25 packer: 

2019/11/05 16:58:25 packer: <nil>
```

My best guess (and forgive me if it's either obvious or a boneheaded hunch) is that Vagrant Cloud expects an upload of a given size but reads an EOF prior to receiving that many bytes.

I noticed these seemingly related, as-yet unresolved issues:
- [#3378](https://github.com/hashicorp/packer/issues/3378)
- #8229 

If this is just my mistake, please let me know what I'm doing wrong. But if it's actually a bug and someone can point me in the right direction as to the nature of the error, I'd be happy to discuss contributing a fix.

Thanks for the time/attention :boom:

**Comment by Peleke** [comment]
**Update**: Looks like the upload completes (more or less) successfully, actually; a new version of the box appears in my Vagrant Cloud UI. 

However, the EOF error causes the build to fail regardless, and the box ends up `unreleased`, forcing a manual release step.

**Comment by SwampDragons** [comment]
I'm doing some old issue cleanup, and this looks like a duplicate of #6615 so I'm closing. We can track the issue there.
